action/register.go

```diff
@@ -0,0 +1,121 @@
+package action
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+
+	"github.com/allegro/vaas-registration-hook/k8s"
+	"github.com/allegro/vaas-registration-hook/vaas"
+)
+
+const (
+	// RegisterName is the CLI name of this action
+	RegisterName = "register"
+
+	// FlagWeight is the weight that a backend in VaaS is initially assigned
+	FlagWeight = "weight"
+	// FlagDC represents datacenter short name as defined in VaaS
+	FlagDC = "dc"
+
+	canaryTag = "canary"
+)
+
+// GetRegisterFlags returns a list of flags available for this action
+func GetRegisterFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.IntFlag{
+			Name:  FlagWeight,
+			Usage: "initial weight of this backend",
+			Value: 1,
+		},
+		cli.StringFlag{
+			Name:  FlagDC,
+			Usage: "datacenter short name as defined in VaaS",
+		},
+	}
+}
+
+// RegisterCLI configures a VaaS client from CLI data and runs register()
+func RegisterCLI(c *cli.Context) error {
```

Comment:
The same as a previous comment, registration based on k8s annotations would be enough imo.